numpy/lib/tests/test_function_base.py

```diff
@@ -2716,6 +2820,28 @@ def test_with_incorrect_minlength(self):
                             "must not be negative",
                             lambda: np.bincount(x, minlength=-1))
 
+        out = np.zeros(5, dtype=int)
+        assert_raises_regex(ValueError,
+                            "cannot be larger than the size of 'out'",
+                            lambda: np.bincount(x, minlength=10, out=out))
+
+        initial = np.zeros(5, dtype=int)
+        assert_raises_regex(ValueError,
+                            "cannot be larger than the size of 'initial'",
+                            lambda: np.bincount(x, minlength=10, 
+                                                initial=initial))
+
+    def test_with_incorrect_types(self):
+        x = np.array([], dtype=int)
+        out = np.empty(6, dtype=float)
+        assert_raises_regex(TypeError,
+                            "Cannot cast",
+                            lambda: np.bincount(x, out=out))
```

Comment:
Should this be an error? In principle, casting is possible. But I can see that this would make things more complicated, so probably not worth it!